### PR TITLE
render workers using cluster chart

### PR DESCRIPTION
**Pull request by anvddriesch**
https://github.com/giantswarm/giantswarm/issues/31416

This pr: 

This migrates all the worker resources to be rendered from the cluster chart.

### Trigger e2e tests

/run cluster-test-suites

### CODE REVIEW

File: helm/cluster-vsphere/README.md [hunk 1]
```diff
@@ -79,6 +79,12 @@ Properties within the `.global.controlPlane` object
 | `global.controlPlane.apiServerPort` | **API server port** - The API server Load Balancer port. This option sets the Spec.ClusterNetwork.APIServerPort field on the Cluster CR. In CAPI this field isn't used currently. It is instead used in providers. In CAPA this sets only the public facing port of the Load Balancer. In CAPZ both the public facing and the destination port are set to this value. CAPV and CAPVCD do not use it.|**Type:** `integer`<br/>**Default:** `6443`|
 | `global.controlPlane.image` | **Node container image**|**Type:** `object`<br/>|
 | `global.controlPlane.image.repository` | **Repository**|**Type:** `string`<br/>**Default:** `"gsoci.azurecr.io/giantswarm"`|
+| `global.controlPlane.machineHealthCheck` | **Machine health check**|**Type:** `object`<br/>|
```
Comment (anvddriesch) [inline helm/cluster-vsphere/README.md:82]: apparently the machine health check was just _there_ previously. Now we have options. The defaults are the same as before.

File: helm/cluster-vsphere/files/etc/ssh/sshd_config [hunk 2]
```diff
@@ -1,16 +0,0 @@
-# DEPRECATED - remove once CP and workers are rendered with cluster chart
```
Comment (anvddriesch) [inline helm/cluster-vsphere/files/etc/ssh/sshd_config:1]: I got rid of everything marked DEPRECATED but let's check if there's more old stuff

File: helm/cluster-vsphere/templates/helpers/_workers.tpl [hunk 3]
```diff
@@ -0,0 +1,11 @@
+{{/*
+Generates template spec for worker machines.
+*/}}
+{{- define "worker-vspheremachinetemplate-spec" -}}
```
Comment (anvddriesch) [inline helm/cluster-vsphere/templates/helpers/_workers.tpl:4]: nothing new here, we just need the spec separated out.

File: helm/cluster-vsphere/templates/vspheremachinetemplate.yaml [hunk 4]
```diff
@@ -17,7 +17,7 @@ spec:
 apiVersion: {{ include "infrastructureApiVersion" . }}
 kind: VSphereMachineTemplate
 metadata:
-  name: {{ include "resource.default.name" $ }}-{{ $nodePoolName }}-{{ include "mtRevision" $c }}
+  name: {{ include "resource.default.name" $ }}-{{ $nodePoolName }}-{{ include "machineTemplateSpec.hash" (dict "data" (include "worker-vspheremachinetemplate-spec" $) "salt" $.Values.cluster.providerIntegration.hashSalt) }}
```
Comment (anvddriesch) [inline helm/cluster-vsphere/templates/vspheremachinetemplate.yaml:20]: the naming function that ensures machines have the same names as referenced inside the cluster chart resources

File: helm/cluster-vsphere/values.schema.json [hunk 5]
```diff
@@ -331,21 +368,36 @@
...
                             "version": "v1beta1"
                         },
+                        "infrastructureMachinePool": {
+                            "group": "infrastructure.cluster.x-k8s.io",
+                            "kind": "VSphereMachineTemplate",
+                            "version": "v1beta1"
+                        },
                         "machineHealthCheckResourceEnabled": true,
-                        "machinePoolResourcesEnabled": false,
+                        "machinePoolResourcesEnabled": true,
```
Comment (anvddriesch) [inline helm/cluster-vsphere/values.schema.json:377]: this just enables rendering worker resources in the cluster chart

**Comment by anvddriesch** [comment]
Please correct me if I'm wrong but in case users have migrated to v0.60.0 there should not be another breaking change, right?
(also I'll add the changelog but please behold the green checkmark first)

**Comment by anvddriesch** [comment]
Please correct me if I'm wrong but in case users have migrated to v0.60.0 there should not be another breaking change, right?
(also I'll add the changelog but please behold the green checkmark first)

File: helm/cluster-vsphere/values.yaml [hunk 6]
```diff
@@ -163,19 +163,31 @@ cluster:
         group: infrastructure.cluster.x-k8s.io
         kind: VSphereCluster
         version: v1beta1
+      infrastructureMachinePool:
+        group: infrastructure.cluster.x-k8s.io
+        kind: VSphereMachineTemplate
+        version: v1beta1
       machineHealthCheckResourceEnabled: true
-      machinePoolResourcesEnabled: false
+      machinePoolResourcesEnabled: true
```
Comment (vxav) [inline helm/cluster-vsphere/values.yaml:171]: Do we need this since we don't use machine pools?

Comment (anvddriesch) [reply helm/cluster-vsphere/values.yaml:171]: We do! The naming is a bit weird but it just refers to worker resources in general.

**Comment by vxav** [comment]
There's quite a lot going on here but nothing jumps at me here.
However I will leave the review to Simon who is much more up to date with the chart changes than I am lately.

**Comment by anvddriesch** [comment]
/run cluster-test-suites TARGET_SUITES=./providers/capv/standard

**Review by glitchcrab — APPROVED** [review]
me likey

**Pull request merged**